src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java

```diff
@@ -184,44 +173,34 @@ protected AbstractAuditLog(Settings settings, final ThreadPool threadPool, final
             log.info("Configured Users to ignore for write compliance events: {}", ignoredComplianceUsersForWrite);
         }
 
-
-
         ignoreAuditRequests = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS, Collections.emptyList());
         if (ignoreAuditRequests.size() > 0) {
             log.info("Configured Requests to ignore: {}", ignoreAuditRequests);
         }
 
-        // check if some categories are invalid
-        for (String event : disabledRestCategories) {
-        	try {
-        		AuditMessage.Category.valueOf(event.toUpperCase());
-        	} catch(Exception iae) {
-        		log.error("Unkown category {}, please check opendistro_security.audit.config.disabled_categories settings", event);
-        	}
-		}
+        this.excludeSensitiveHeaders = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true);
+    }
 
-        // check if some categories are invalid
-        for (String event : disabledTransportCategories) {
-            try {
-                AuditMessage.Category.valueOf(event.toUpperCase());
-            } catch(Exception iae) {
-                log.error("Unkown category {}, please check opendistro_security.audit.config.disabled_categories settings", event);
-            }
+    private static List<String> getConfigList(final Settings settings,
```

Comment:
Consider moving to `AuditCategory`.

Reply:
This is generic reading of config values and feel it should not be part of AuditCategory. Will be moving this to AuditConfig class

Reply:
`AuditConfig` or `AuditCategory`, it does not belong to `AbstractAuditLog`, IMO.

Reply:
I plan to refactor these audit configurations into a separate AuditConfig class to represent all audit log configuration (basically abstracting it out of AbstractAuditLog). 
Let me create a milestone in Github for having better clarity of changes I want to make.

Reply:
As this is a new function, it will be better to put it to `AuditCategory` compared to `AbstractAuditLog`. If necessary, please add TODO comment.

Reply:
getConfigList is something that will be reused by other configs like ignoredAuditUsers, ignoredComplianceUsersForRead, ignoredComplianceUsersForWrite. And they all should go into an AuditConfig class.

Reply:
This is going to be fixed with #304 306

Reply:
Can you introduce `AuditConfig` class in this PR with a single method and extend it in #306?

Reply:
Why add here and not in #306 ?

Reply:
It is already used in this PR.

Reply:
Sorry I m confused. What is already used in this PR?

Reply:
`getConfigList()` method

Reply:
Are you saying to include all AuditConfig changes in this PR itself ?

Reply:
#306 should now have only #323 and #312 changes if that what you are asking for

Reply:
No, only `getConfigList()`. Note that I have not looked at the other PR yet.

Reply:
What is the value in adding just that in this ? Am I missing something?

Reply:
It will finalize this change and will not require more changes in the follow-up PRs.

Reply:
I don't think we should assume any change is finalized and will not undergo more changes in the future PRs.
This function is just pulled out of existing code in AbstractAuditLog class into a method. 
I don't think it's a good idea to add incomplete AuditConfig just for not reviewing this change again as the new diff will have changes. 

I would recommend reviewing #306 instead which is a more complete solution of what you are looking for AuditConfig + AuditCategory.

Reply:
Once `getConfigList()` is moved to `AuditConfig` what other changes do you plan to apply to it?

Reply:
It may not have any new change but I m not seeing any value in adding incomplete AuditConfig class here. 
I feel I m missing your point. Are you saying getConfigList does not need to reviewed again?

Reply:
>It may not have any new change but I m not seeing any value in adding incomplete AuditConfig class here.

`AuditConfig` is likely to evolve over time, so it will never be complete. 

>I feel I m missing your point. Are you saying getConfigList does not need to reviewed again?

It may require a review in case the follow-up PRs need to introduce other changes to `getConfigList()` that are not necessary for this PR, but it will be easy to review. In addition, it will also simplify effort if somebody needs to look at git history for the function.

Reply:
Since you are reviewing #306  we can close this PR

Reply:
Please don't close this PR. It is almost ready to go.

Reply:
#306 should have the change you were looking with getConfigList()

Reply:
I'd prefer not to squash commits in #306, so #306 won't address my concern.

Reply:
So confused. What is your concern ? and why not squash commits?

Reply:
Doesn't #306 address your concern with `getConfigList()` ?

Reply:
I'd prefer to keep `AuditCategory` changes separate from other changes introduced in #306.

Reply:
It is the same commit that is cherry-picked 0cb732f . So changes are as it is

Reply:
Without squashing all commits together #306 does not address my concern regarding `getConfigList()` and I'd prefer not to squash commits to keep `AuditCategory` changes separate from other changes introduced in #306.

Reply:
You concern of getConfigList() is resolved with #306 .
Why do you prefer to keep this change separate. How does it matter ?

Reply:
If you are worried about any git history and having an atomic merge step (auditconfig with getConfigList), then going forward with #306 is the right way.

Reply:
With 2 separate commits, it will be much easier to understand what was changed and why and how changes are related for anyone who is not involved in the current PR in the future.

Reply:
I think you also agree that `getConfigList()` does not belong to `AbstractAuditLog` and it should be part of `AuditConfig` class. Should this method previously exist as part of `AbstractAuditLog` it would be OK to move it in subsequent PR. As it is a newly introduced method, I would prefer to put it in the right place when it is introduced.

Reply:
I agree getConfigList() belongs to AuditConfig class and thats exactly what #306 does. 
At the same time, I dont believe this must exist in some incomplete class that may be introduced in a future PR.

Squashing the commits of #306 would look like everything was included in a single PR/commit keeping git history clean. 
Well, since this PR is not merged yet, this is not the right place when it was introduced and the right place would be #306

Reply:
Squashing all commits in #306 would create a single commit with a large number of changes that logically can be split into two commits. This will not only mess up the git history but more importantly, it will create a huge commit that will be harder to understand.